DevelopGitHub/GitHubExtension/GitHubExtension.Notes.WebApi/Controllers/NoteController.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Threading.Tasks;
+using System.Web.Http;
+using GitHubExtension.Notes.WebApi.Commands;
+using GitHubExtension.Notes.WebApi.Constants;
+using GitHubExtension.Notes.WebApi.Mappers;
+using GitHubExtension.Notes.WebApi.Queries;
+using GitHubExtension.Notes.WebApi.ViewModels;
+
+namespace GitHubExtension.Notes.WebApi.Controllers
+{
+    public class NoteController : ApiController
+    {
+        private INoteCommands commands;
+        private INoteQueries queries;
+
+        public NoteController(INoteCommands commands, INoteQueries queries)
+        {
+            this.commands = commands;
+            this.queries = queries;
+        }
+
+        [Route(NotesRouteConstants.GetNoteRoute)]
+        [HttpGet]
+        public async Task<IHttpActionResult> GetNote([FromUri] string collaboratorId)
+        {
+            var userId = IdentityExtensions.GetUserId();
+            var note = await queries.GetNote(userId, collaboratorId);
+            if (note == null)
+            {
+                return NotFound();
+            }
+            var noteModel = note.ToNoteViewModel();
+            return Ok(noteModel);
+        }
+
+        [Route(NotesRouteConstants.CreateNoteRoute)]
+        [HttpPost]
+        public async Task<IHttpActionResult> CreateNote(AddNoteModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var userId = IdentityExtensions.GetUserId();
+            var noteModel = model.AddUserIdToModel(userId);
+            if (noteModel == null)
```

Comment:
I need to check userId for null before mapping, made so and commited changes. I do not see a reason to check a model which comes from UI, because it is validated with FluentValidation and if there are some problems it returns a BadRequest with error message.